baseplate.go

```diff
@@ -77,17 +77,42 @@ type Server interface {
 	Serve() error
 }
 
-// Serve runs the given Server until it is given an external shutdown signal
-// using runtimebp.HandleShutdown to handle the signal and shut down the
-// server gracefully.  Returns the (possibly nil) error returned by "Close" or
+// ServeArgs provides a list of arguments to Serve.
+//    server: Mandatory. Indicates the Server that should be run until
+//            receiving a shutdown signal.
+//    preShutdown: Optional. Indicates a slice of io.Closers that should
+//						be gracefully shut down before the server upon receipt of a
+//            shutdown signal.
+//    preShutdown: Optional. Indicates a slice of io.Closers that should
+//						be gracefully shut down after the server upon receipt of a
+//            shutdown signal.
```

Comment:
I'm not actually sure how formatting (and tabs vs spaces) works in whatever utility is generating API documentation from this, so I'd love feedback on that!

Reply:
Interestingly it shows up properly in the `/files` endpoint but improperly in this snippet.

I used spaces when laying out the comment but I guess `go fmt` might have tabified them.

Reply:
I believe tabs make things look like code and yiou should use spaces.  Also, you don't want to do this sort of spacing, i.e.:

```go
// ServeArgs provides a list of arguments to Serve.
// server: Mandatory. Indicates the Server that should be run until
// receiving a shutdown signal.
```